naga/src/front/wgsl/parse/mod.rs

```diff
@@ -2762,14 +2839,34 @@ impl Parser {
         }
     }
 
-    pub fn parse<'a>(&mut self, source: &'a str) -> Result<ast::TranslationUnit<'a>, Error<'a>> {
+    pub fn parse<'a>(
+        &mut self,
+        source: &'a str,
+        options: &Options,
+    ) -> Result<ast::TranslationUnit<'a>, Error<'a>> {
         self.reset();
 
-        let mut lexer = Lexer::new(source);
+        let mut lexer = Lexer::new(source, options.parse_comments);
         let mut tu = ast::TranslationUnit::default();
         let mut enable_extensions = EnableExtensions::empty();
         let mut diagnostic_filters = DiagnosticFilterMap::new();
 
+        if options.parse_comments {
+            // Parse module comments.
+            let mut comments = Vec::new();
+
+            fn peek_any_next<'a>(lexer: &'a Lexer) -> (Token<'a>, Span) {
+                let mut cloned = lexer.clone();
+                let token = cloned.next_until(|_| true, false);
+                token
+            }
+            while let (Token::CommentDocModule(_), span) = peek_any_next(&lexer) {
```

Comment:
We could add a `next_if` method on the lexer for this, making the code easier to look at.